main/src/library/File.flix

```diff
@@ -840,4 +840,100 @@ namespace File {
                 Err(getMessage(ex))
         }
 
+    ///
+    /// Returns a list naming the files and directories in the directory `f`.
+    ///
+    /// Does not recursively traverse the directory.
+    ///
+    pub def list(f: String): Result[List[String], String] & Impure =
+        try {
+            import new java.io.File(String): ##java.io.File & Impure as newFile;
+            import java.io.File.list(): Array[String] & Impure;
```

Comment:
This function : "Returns null if this abstract pathname does not denote a directory, or if an I/O error occurs." I should probably check for this an return an error instead